libobs: Fix source profiling inactive sources

exeldro opened the pull request:
### Description
fix source profiling inactive sources

### Motivation and Context
When filters are disabled the tick takes time, but the cpu and gpu render do not, so I would like to see that in `profiler_result_t`.
Before it would show the last status before it was disabled, now the cpu and gpu go to 0 in 5 seconds

### How Has This Been Tested?
With https://github.com/exeldro/obs-source-profiler by enabling and disabling a filter.

### Types of changes
- Bug fix (non-breaking change which fixes an issue) 

### Checklist:
- [x] My code has been run through [clang-format](https://github.com/obsproject/obs-studio/blob/master/.clang-format).
- [x] I have read the [**contributing** document](https://github.com/obsproject/obs-studio/blob/master/CONTRIBUTING.rst).
- [x] My code is not on the master branch.
- [x] The code has been tested.
- [x] All commit messages are properly formatted and commits squashed where appropriate.
- [x] I have included updates to all appropriate documentation.

derrod reviewed (commented):
I think originally I did it this way to avoid a bit of overhead for inactive sources and then just hide stuff in the UI, but I suppose this also makes sense.

BTW if you release https://github.com/exeldro/obs-source-profiler it would be nice to be credited since it seems to be based on my original UI prototype branch :)

exeldro commented:
@derrod I made your source profiler UI code into a plugin and changed some things to make testing easier for me. I now updated the readme to credit you. I will not make releases for mine if you plan to release a plugin yourself. Otherwise I will also credit you on the OBS forum resource page.

derrod commented:
> @derrod I made your source profiler UI code into a plugin and changed some things to make testing easier for me. I now updated the readme to credit you. I will not make releases for mine if you plan to release a plugin yourself. Otherwise I will also credit you on the OBS forum resource page.

Ab I see. I still plan on releasing a standalone plugin once 31.0 beta is out, but wouldn't mind somebody else also releasing a version with improved UI 😅
